lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -136,6 +134,9 @@ export class PublicClientApplication extends ClientApplication implements IPubli
             const silentAuthClient = await this.createSilentFlowClient(serverTelemetryManager, silentRequest.authority);
             return silentAuthClient.acquireCachedToken(silentRequest);
         } catch (e) {
+            if (this.embeddedApp && this.embeddedApp.brokerConnectionEstablished) {
```

Comment:
Why are we not checking it before the catch instead of letting silent fail?

Reply:
Tokens will be cached on the embedded app, so we only need to check if brokering is enabled before making the network request.